[patch] [bug] CEECORE-686 Windows memfs fixed

scrajesh opened the pull request:
* Added custom (optional) path property to ServeIndex(). 
* It's mainly used to set Posix Path methods/props for MemoryFileSystem

scrajesh committed:
making Path object to be within ServeIndex object

This will create new version of Path(either native or posix) for every call to serveIndex()

scrajesh committed:
Move helper functions inside ServeIndex()

This will help to scope the custom pathLib determined thru options.

scrajesh committed:
[patch] [bug] CEECORE-686 Windows memfs fixed

Added custom optinal path property to ServeIndex(). It's used to send posix notation of Path methods when the MemoryFileSystem is in place.

scrajesh committed:
[chore] cleanup logs

scrajesh committed:
Merge remote-tracking branch 'upstream/master'

scrajesh committed:
[chore] fixed failing unit tests

now custom response-type handler can be passed via options

scrajesh committed:
[chore] fixing lint issues

jchip commented on index.js:
```diff
@@ -91,453 +76,471 @@ function serveIndex(root, options) {
+  // default Stylesheet
+  var defaultStylesheet = join(__dirname, 'public', 'style.css');
+  // default Template
+  var defaultTemplate = join(__dirname, 'public', 'directory.html');
```
these two are for the module to find its own static file from disk, therefore, they should use native path.join.

scrajesh replied:
corrected and moved it back to global scope.

jchip commented on test/test.js:
```diff
@@ -485,15 +485,15 @@ describe('serveIndex(root)', function () {
 
   describe('when using custom handler', function () {
```
can we ensure existing tests pass as is please?  ie: make no changes to tests.

scrajesh replied:
removed the optional responseHandler and used same serveIndex.html/plain/json object notation for custom handlers.

jchip commented on index.js:
```diff
@@ -91,453 +76,471 @@ function serveIndex(root, options) {
+  /**
+   * Respond with text/html.
+   */
+  function html(req, res, files, next, dir, showUp, displayIcons, path, view, template, stylesheet, filesystem) {
```
nit: moving so much code around cause a huge diff.  If possible, please move the code to end so original code at line 105 remains where they were to help with reduce code changes.

scrajesh replied:
named the returning function and retained it as per the original order.
moved the return statement to the end of the function.
hope this reduces the difference considerably.

scrajesh committed:
[chore] reverting changes in the tests

removed the optional requestHandler prop and instead used the previous custom prop notation for the response handlers

scrajesh committed:
[chore] PR feedback

changing default template/stylesheet back to use native path.join() irrespective of the platform.
As, they are the actual files on the disk.

scrajesh committed:
[chore] PR feedback

Named the return function and moved the return statement to the end.
It shoiuld maintain the huge code diffenrence.

scrajesh committed:
[fix] Following Class structure rather a function with closure
